firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/common/CrashlyticsControllerTest.java

```diff
@@ -842,6 +842,21 @@ public void testUploadDisabledThenEnabled() throws Exception {
     Task<Void> task = controller.submitAllReports(1.0f, testSettingsDataProvider.getAppSettings());
 
     arbiter.setCrashlyticsDataCollectionEnabled(true);
+    assertTrue(arbiter.isAutomaticDataCollectionEnabled());
+
+    when(mockEditor.putBoolean(PREFS_KEY, false)).thenReturn(mockEditor);
+    when(mockPrefs.getBoolean(PREFS_KEY, true)).thenReturn(false);
+    arbiter.setCrashlyticsDataCollectionEnabled(false);
+    assertFalse(arbiter.isAutomaticDataCollectionEnabled());
+
+    when(mockPrefs.contains(PREFS_KEY)).thenReturn(false);
```

Comment:
Can you split this up into separate tests that express the conditions they check?